src/app/modules/todo/services/todo.service.ts

```diff
@@ -30,32 +37,47 @@ export class TodoService {
       .collection('Todo', (ref) => ref.where('title', '==', todoTitle))
       .snapshotChanges()
       .pipe(
-        map((result) => {
+        switchMap((result) => {
+          return this.authService.checkAuthAPI(result);
+        }),
+        map((result: any[]) => {
           const listTodo: Todo[] = this.formatListTodo(result);
           const itemTodo = listTodo.shift();
           return itemTodo;
         })
       );
   }
 
-  public add(newTodo: Todo): void {
-    const todo: Todo = this.createTodo(
-      newTodo.title,
-      newTodo.content,
-      newTodo.creator,
-      newTodo.deadLine
-    );
-    this.firestore.collection('Todo').add(todo);
+  public async add(newTodo: Todo): Promise<StatusRequest> {
+    const checkResult = this.authService.checkExpirationToken();
+    if (checkResult.status) {
+      const todo: Todo = this.createTodo(
+        newTodo.title,
+        newTodo.content,
+        newTodo.creator,
+        newTodo.deadLine
+      );
+      await this.firestore.collection('Todo').add(todo);
+    }
+    return checkResult;
   }
 
-  public updateTodo(todo: Todo): void {
-    const todoId = todo.id;
-    delete todo.id;
-    this.firestore.collection('Todo').doc(todoId).update(todo);
+  public async updateTodo(todo: Todo): Promise<StatusRequest> {
+    const checkResult = this.authService.checkExpirationToken();
+    if (checkResult.status) {
+      const todoId = todo.id;
+      delete todo.id;
+      await this.firestore.collection('Todo').doc(todoId).update(todo);
+    }
+    return checkResult;
   }
 
-  public delete(id: string): void {
-    this.firestore.collection('Todo').doc(id).delete();
+  public async delete(id: string): Promise<StatusRequest> {
+    const checkResult = this.authService.checkExpirationToken();
+    if (checkResult.status){
```

Comment:
make a whitespace before `{`. Please change vscode config so you only need to hit a short key to format your file .